app/cothorityd/cothorityd.go

```diff
@@ -1,127 +1,400 @@
+const DefaultServerConfig = "config.toml"
 
-// Initialize before 'init' so we can directly use the fields as parameters
-// to 'Flag'
-func init() {
-	flag.StringVar(&ConfigFile, "config", "cothorityd.toml", "which config-file to use")
-	flag.IntVar(&debugVisible, "debug", 1, "verbosity: 0-5")
-}
+// DefaultGroupFile is the default name of a group definition file
+const DefaultGroupFile = "group.toml"
+
+// Version of this binary
+const Version = "1.1"
 
-// Main starts the host and will setup the protocol.
 func main() {
 
 	cliApp := cli.NewApp()
 	cliApp.Name = "Cothorityd server"
 	cliApp.Usage = "Serve a cothority"
-	cliApp.Version = "1.0"
-	cliApp.Flags = []cli.Flag{
+	cliApp.Version = Version
+	serverFlags := []cli.Flag{
 		cli.StringFlag{
 			Name:  "config, c",
-			Value: "cothorityd.toml",
-			Usage: "config-file for the server",
+			Value: getDefaultConfigFile(),
+			Usage: "Configuration file of the server",
 		},
 		cli.IntFlag{
 			Name:  "debug, d",
 			Value: 1,
 			Usage: "debug-level: 1 for terse, 5 for maximal",
 		},
 	}
-	cliApp.Action = func(c *cli.Context) {
-		config := c.String("config")
-		dbg.SetDebugVisible(c.Int("debug"))
-		dbg.Lvl1("Starting cothority daemon", cliApp.Version)
-		startCothorityd(config)
+
+	cliApp.Commands = []cli.Command{
+		{
+			Name:    "setup",
+			Aliases: []string{"s"},
+			Usage:   "Setup the configuration for the server (interactive)",
+			Action: func(c *cli.Context) error {
+				if c.String("config") != "" {
+					stderrExit("[-] Configuration file option can't be used for the 'setup' command")
+				}
+				if c.String("debug") != "" {
+					stderrExit("[-] Debug option can't be used for the 'setup' command")
+				}
+				interactiveConfig()
+				return nil
+			},
+		},
+		{
+			Name:  "server",
+			Usage: "Run the cothority server",
+			Action: func(c *cli.Context) {
+				runServer(c)
+			},
+			Flags: serverFlags,
+		},
+	}
+	cliApp.Flags = serverFlags
+	// default action
+	cliApp.Action = func(c *cli.Context) error {
+		runServer(c)
+		return nil
 	}
 
 	cliApp.Run(os.Args)
 
 }
 
-func startCothorityd(configName string) {
-	if _, err := os.Stat(ConfigFile); os.IsNotExist(err) {
-		// the config file does not exists, let's create it
-		config, fname, err := config.CreateCothoritydConfig(configName)
+func runServer(ctx *cli.Context) {
+	// first check the options
+	dbg.SetDebugVisible(ctx.Int("debug"))
+	config := ctx.String("config")
+
+	if _, err := os.Stat(config); os.IsNotExist(err) {
+		dbg.Fatalf("[-] Configuration file does not exists. %s", config)
+	}
+	// Let's read the config
+	_, host, err := c.ParseCothorityd(config)
+	if err != nil {
+		dbg.Fatal("Couldn't parse config:", err)
+	}
+	host.ListenAndBind()
+	host.StartProcessMessages()
+	host.WaitForClose()
+
+}
+
+// CreateCothoritydConfig will ask through the command line to create a Private / Public
+// key, what is the listening address
+func interactiveConfig() {
```

Comment:
Usually I would agree but here everything is closely related and it's a *one shot* function. If I split it, there would be a part about writing the config file, generating the keys and asking for (maybe 2) ip addresses, but none of these functions would be called elsewhere, because they are only needed here.
But yes, everything is splittable anyway, so I can try to split it up :D